ci/setup-build-test-gdk.ps1

```diff
@@ -30,33 +69,71 @@ Start-Event "setup-gdk" "command"
 & $PSScriptRoot"\setup-gdk.ps1" -gdk_path "$gdk_in_engine" -msbuild_path "$msbuild_exe"
 Finish-Event "setup-gdk" "command"
 
-# Build the testing project
-Start-Event "build-project" "command"
-& $PSScriptRoot"\build-project.ps1" `
-    -unreal_path "$unreal_path" `
-    -test_repo_branch "$test_repo_branch" `
-    -test_repo_url "$test_repo_url" `
-    -test_repo_uproject_path "$build_home\TestProject\$test_repo_relative_uproject_path" `
-    -test_repo_path "$build_home\TestProject" `
-    -msbuild_exe "$msbuild_exe" `
-    -gdk_home "$gdk_home" `
-    -build_platform "$env:BUILD_PLATFORM" `
-    -build_state "$env:BUILD_STATE" `
-    -build_target "$env:BUILD_TARGET"
-Finish-Event "build-project" "command"
-
-# Only run tests on Windows, as we do not have a linux agent - should not matter
-if ($env:BUILD_PLATFORM -eq "Win64" -And $env:BUILD_TARGET -eq "Editor" -And $env:BUILD_STATE -eq "Development") {
-  Start-Event "test-gdk" "command"
-  & $PSScriptRoot"\run-tests.ps1" `
-      -unreal_editor_path "$unreal_path\Engine\Binaries\Win64\UE4Editor.exe" `
-      -uproject_path "$build_home\TestProject\$test_repo_relative_uproject_path" `
-      -test_repo_path "$build_home\TestProject" `
-      -log_file_path "$PSScriptRoot\TestResults\tests.log" `
-      -test_repo_map "$test_repo_map"
-  Finish-Event "test-gdk" "command"
-
-  Start-Event "report-tests" "command"
-  & $PSScriptRoot"\report-tests.ps1" -test_result_dir "$PSScriptRoot\TestResults" -target_platform "$env:BUILD_PLATFORM"
-  Finish-Event "report-tests" "command"
+class CachedProject {
+  [ValidateNotNullOrEmpty()][string]$test_repo_url
+  [ValidateNotNullOrEmpty()][string]$test_repo_branch
+
+  CachedProject([string] $test_repo_url, [string] $test_repo_branch) {
+    $this.test_repo_url = $test_repo_url
+    $this.test_repo_branch = $test_repo_branch
+  }
 }
+
+$projects_cached = @()
+
+foreach ($test in $tests) {
+  $test_repo_url = $test.test_repo_url
+  $test_repo_branch = $test.test_repo_branch
+  $test_repo_relative_uproject_path = $test.test_repo_relative_uproject_path
+  $test_repo_map = $test.test_repo_map
+  $test_project_root = $test.test_project_root
+  $test_results_dir = $test.test_results_dir
+  $tests_path = $test.tests_path
+  $run_with_spatial = $test.run_with_spatial
+
+  $project_is_cached = $False
```

Comment:
Why is this on a different line

Reply:
The above is copying the object into local variables, while this is just an unrelated local variable